be/src/util/debug/query_trace.h

```diff
@@ -0,0 +1,78 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#include "util/debug/query_trace_impl.h"
+// A lightweight tool for analyzing query performance under the pipeline engine.
+// The basic background and usage can refer to the description of https://github.com/StarRocks/starrocks/pull/7649
+// How to add trace point?
+// Currently only duration events and async events are supported.
+// An example of adding duration event is as follows
+// void doSomething() {
+//     SET_THREAD_LOCAL_QUERY_TRACE_CONTEXT(query_trace, fragment_instance_id, driver_ptr);
+//     QUERY_TRACE_BEGIN("category", "name");
+//     process_function();
+//     QUERY_TRACE_END("category", "name");
+// }
+// or just use QUERY_TRACE_SCOPED
+// void process_function () {
+//     QUERY_TRACE_SCOPED("category", "name");
+//     do something
+// }
+// void doSomething() {
+//     SET_THREAD_LOCAL_QUERY_TRACE_CONTEXT(query_trace, fragment_instance_id, driver_ptr);
+//     process_function();
+// }
+// It should be noted that these QUERY_TRACE_* macros rely on thread local trace context to read some information,
```

Comment:
Why do we need a thread-local context?

Reply:
> Why do we need a thread-local context?

in order to make macros in other places more convenient to use, we need this thread-local var to pass some common information